lib/manager/pip_requirements/extract.ts

```diff
@@ -53,6 +53,7 @@ export function extractPackageFile(
   }
   registryUrls = registryUrls.concat(extraUrls);
 
+  const pkgRegex = regEx(`^(${packagePattern})$`);
```

Comment:
Can't we make the other parts optional and use a single regex?

Reply:
I wanted to keep the changes as low risk as possible because we know the existing works reliably. Prefer we merge this and then someone can refactor later if really desired